Source/Core/BareMetal/L6App/Loader.cpp

```diff
@@ -137,6 +137,19 @@ ErrorManagement::ErrorType Loader::Configure(StructuredDataI &data, StreamI &con
     }
     if (ret.ErrorsCleared()) {
         ret.fatalError = !parsedConfiguration.MoveToRoot();
+        //We need items which exist outside the actual application to sometimes know the first state such as the statemachine
```

Comment:
It seems you would like to write in the StructuredDataI the first state that was provided by the user, so that it can be later used by other components.

I believe this sort of behaviour should be implemented outside the context of the MARTe application, e.g. by patching the configuration file before loading with  _sed_ or equivalent.